blockchain/chain/src/store/chain_store.rs

```diff
@@ -437,6 +450,78 @@ where
     })
 }
 
+/// Attempts to deserialize to unsigend message or signed message and then returns it at as a message trait object
+pub fn get_chain_message<DB>(db: &DB, key: &Cid) -> Result<BoxMessage, Error>
+where
+    DB: BlockStore,
+{
+    let value = db
+        .read(key.key())?
+        .ok_or_else(|| Error::UndefinedKey(key.to_string()))?;
+    if let Ok(message) = from_slice::<UnsignedMessage>(&value) {
+        Ok(Box::new(message))
+    } else {
+        let signed_message: SignedMessage = from_slice(&value)?;
+        Ok(Box::new(signed_message))
+    }
+}
+
+// given a tipset this function will return all messages as a trait object
+pub fn messages_for_tipset<DB>(db: &DB, ts: &Tipset) -> Result<Vec<BoxMessage>, Error>
+where
+    DB: BlockStore,
+{
+    let mut applied: HashMap<Address, u64> = HashMap::new();
+    let mut balances: HashMap<Address, BigInt> = HashMap::new();
+    let state = StateTree::new_from_root(db, ts.parent_state())?;
+
+    // message to get all messages for block_header into a single iterator
+    let mut get_message_for_block_header = |b: &BlockHeader| -> Result<Vec<BoxMessage>, Error> {
+        let (unsigned, signed) = block_messages(db, b)?;
+        let unsigned_box = unsigned.into_iter().map(|s| Box::new(s) as BoxMessage);
+        let signed_box = signed.into_iter().map(|s| Box::new(s) as BoxMessage);
+
+        let mut messages = Vec::new();
+        for message in unsigned_box.chain(signed_box) {
+            let from_address = message.from();
```

Comment:
Seems we are skipping the step of preloading the address here, is there a specific reason for this? Should at least add a TODO maybe?

Reply:
If we are speaking about this step :  https://github.com/filecoin-project/lotus/blob/c4de2373fd13c909876cf459e59ead7749b9939d/chain/store/store.go#L718 that is moved further down in the loop as I saw no reason as to why they had it in a closure the preload essentially adds it to the hashmap which is done further in the loop